submissions/JOSEF/Alt-F4 - JOSEF 0605.md

```diff
@@ -0,0 +1,158 @@
+
+When all the cells in a new ring are built and enough materials are available in the logistic network, the builder train starts going through them one by one. It only contains a few materials and a BP book that is unloaded into a deployer chest. There are a few tricks involved for cycling through the blueprint book. The full process is too long to describe here, but let’s look at the start as an example. All the cells start with globally connected roboports but in order to check for ores (as described in the next paragraph) I will need uncoupled roboports, so that's the first thing to do:
+
+- BP 1 is a deconstruction planner for roboports. The initial roboports (globally connected) in a cell output *“U-235=1”* (because there’s a chest with exactly 1 U-235 in a chest in the brain). So initially there is a *"U-235=4"* signal from the four roboports in this local circuit network. Once all four roboports are deconstructed (and thereby *"U-235=0"*), a combinator set to *“U-235=0 → construction bot = 1”* triggers, leading to the next blueprint being deployed.
+
+- BP 2 builds a roboport in the center of the cell which is uncoupled from the network. The pre-built belt starts feeding 50 construction bots from the train into it. This new roboport is set to “read robot statistics” and a *“T=50 → construction bot =1”* combinator triggers BP 3 once 50 bots were fed into the now uncoupled network.
+
+- BP 3 deconstructs the bot belt and immediately triggers BP 4, which will start the "resource finding" process described in the next paragraph.
+
+Generally, for cycling through blueprints you‘ll want to find some entity that can output circuit signals and use that entity being built/deconstructed as an indicator for the next step. In addition, you can use "idle construction bots (*T=Z*)" or "item X available in logistic storage" to make sure that the building process is actually complete.
+
+As an example for what kinds of problems you can run into if you're not careful or use time-dependent conditions: As JOSEF started growing larger, it sometimes took a bit of time for bots to come and build the uncoupled roboport. If (for reasons that will appear later) 200 belts were unloaded before that roboport was built, the order of blueprints got messed up, the thing started building belts for mining without deconstructing the bot belt, then dedicated the cell as “no mining”. It was an absolute mess and lead to the factory getting stuck.
+
+## Resource Finder
+
+![Checking for ore patches and placing blueprints accordingly](https://i.imgur.com/NFgi72P.mp4)
+
+Now for one of the crucial bits: How to find resources? There is no way of reading "ore on ground" in vanilla (or any of the mods I'm aware of), but it’s actually really simple once you have uncoupled roboport networks:
```

Comment:
This mod by Earendel can scan for resources :P 
https://mods.factorio.com/mod/aai-programmable-vehicles

Reply:
Okay, I'm not surprised. There's a mod for everything after all :D